### ISSUE TITLE
Wrong line of exception

### ISSUE THREAD

**Issue by CyberMaxRu** [FirebirdSQL/firebird#6895]
SQL:
```
CREATE EXCEPTION EXCEPTION_1 'Cancel';
SET TERM ^ ;

CREATE OR ALTER PROCEDURE TEST_SP
AS
BEGIN
    IF (1 = 1) THEN
    BEGIN
    END

    EXCEPTION EXCEPTION_1;
END^

SET TERM ; ^
```

execute procedure test_sp;
Statement failed, SQLSTATE = HY000
exception 11
-EXCEPTION_1
-Cancel
-At procedure 'TEST_SP' line: 4, col: 5

Line of error - 4. But here no exception.

Commenting IF:
```
SET TERM ^ ;

CREATE OR ALTER PROCEDURE TEST_SP
AS
BEGIN
    --IF (1 = 1) THEN
    BEGIN
    END

    EXCEPTION EXCEPTION_1;
END^

SET TERM ; ^
```

execute procedure test_sp;
Statement failed, SQLSTATE = HY000
exception 11
-EXCEPTION_1
-Cancel
-At procedure 'TEST_SP' line: 8, col: 5

Here line of exception is correct - 8.

**Comment by hvlad** [comment]
I can't reproduce it with current builds of fb3 and fb4.

**Comment by hvlad** [comment]
It was already fixed at #6624
